src/SearchResults/index.js

```diff
@@ -659,20 +659,30 @@ function extractNormalizedFacetValues(results, attribute) {
 }
 
 /**
- * Sort nodes of a hierarchical facet results
+ * Sort nodes of a hierarchical or disjunctive facet results
  * @private
- * @param {HierarchicalFacet} node node to upon which we want to apply the sort
+ * @param {function} sortFn
+ * @param {HierarchicalFacet|Array} node node to upon which we want to apply the sort
+ * @param {string[]} names attribute names
+ * @param {number} [level=0] current index in the names array
  */
-function recSort(sortFn, node) {
+function recSort(sortFn, node, names, level) {
+  level = level || 0;
+
+  if (Array.isArray(node)) {
+    return sortFn(node, names[level]);
+  }
+
   if (!node.data || node.data.length === 0) {
     return node;
   }
 
   var children = node.data.map(function(childNode) {
-    return recSort(sortFn, childNode);
+    return recSort(sortFn, childNode, names, level + 1);
   });
-  var sortedChildren = sortFn(children);
-  var newNode = merge({}, node, {data: sortedChildren});
+  var sortedChildren = sortFn(children, names[level]);
+  var newNode = merge({}, node);
+  newNode.data = sortedChildren;
```

Comment:
this change is because merge would otherwise cause hidden items to reappear

Reply:
I'm not sure to understand. That's worth a comment imo.

Reply:
used defaultsPure which has the expected behaviour without hack